forum/qa-plugin/report-reason/script.js

```diff
@@ -1,32 +1,51 @@
 document.addEventListener("DOMContentLoaded", () => {
     
     const flagboxPopup = document.querySelector("#flagbox-popup");
-    const erorrPopupHtml = document.querySelector('.html-error');
     const flagButton = document.querySelector('.qa-form-light-button-flag');
     flagButton.type = "button";
+    const wrap = document.querySelector('.qa-flag-reasons-wrap');
 
+    function showPopup() {
+        flagboxPopup.classList.remove("hide");
+        flagboxPopup.classList.add("show");
+    }
+    function hidePopup() {
+        flagboxPopup.classList.remove("show");
+        flagboxPopup.classList.add("hide");
+    }
+
+    wrap.addEventListener("click", (e) => {
+
+    e.stopPropagation();
+
+    });
 
     flagButton.addEventListener("click", () => {
+        showPopup();    
+
+        flagboxPopup.addEventListener("click", () => {
```

Comment:
Rozumiem, że na każdy klik we **flagButton** podpinasz się pod klika na **flagboxPopup**? Jeśli tak, to każde kolejne kliknięcie zwielokrotni wywołanie `hidePopup()`. Zatem po pierwsze - jeśli chcesz podpiąć się pod coś raz, to zapodaj do listenera trzeci parametr jako obiekt z flagą `once` na **true**, albo odłączaj go sam przez `removeEventListener` (wtedy musisz zachować sobie referencje do metody obsługującej zdarzenie).